crates/task-impls/src/events.rs

```diff
@@ -39,7 +43,11 @@ pub enum HotShotEvent<TYPES: NodeType> {
     /// Send a quorum vote to the next leader; emitted by a replica in the consensus task after seeing a valid quorum proposal
     QuorumVoteSend(QuorumVote<TYPES>),
     /// Send a DA proposal to the DA committee; emitted by the DA leader (which is the same node as the leader of view v + 1) in the DA task
-    DAProposalSend(Proposal<TYPES, DAProposal<TYPES>>, TYPES::SignatureKey),
+    DAProposalSend(
+        Proposal<TYPES, DAProposal<TYPES>>,
+        TYPES::SignatureKey,
+        usize,
+    ),
```

Comment:
House-cleaning note for the future: for complex enum variants we might consider embedded structs to give each item a name. Or, even better, we could make a stand-alone struct for complex variants. https://stackoverflow.com/a/70321820